### ISSUE TITLE
RangeError: Maximum call stack size exceeded

### ISSUE THREAD

**Issue by MikeAwie** [Typescript-TDD/ts-auto-mock#276]
### Subject of the issue
I am trying to create a mock inside of another mock (in the default values) but getting Maximum call stack size exceeded error.
The classes I am trying to mock are sequelize models.

### Your environment
* ts-auto-mock version: 1.5.5
* typescript version: 3.3.3333
* node version: 12.15.0
* npm version: 6.13.4
* sequelize version:5.18.1

### Steps to reproduce
Call the createMock function with a sequelize model and give it a default value in which you call createMock again:
`const mock = createMock<Model1>({
      property: createMock<Model2>()
  })`

### Expected behavior
Both mocks should work as intended and throw no errors

### Actual behavior
Tell us what happens instead
RangeError: Maximum call stack size exceeded is thrown
`RangeError: Maximum call stack size exceeded
      at node_modules/ts-auto-mock/merge/index.js:1:38397
      at t (node_modules/ts-auto-mock/merge/index.js:1:41229)
      at ko (node_modules/ts-auto-mock/merge/index.js:1:41155)
      at node_modules/ts-auto-mock/merge/index.js:1:41271
      at node_modules/ts-auto-mock/merge/index.js:1:38394
      at t (node_modules/ts-auto-mock/merge/index.js:1:41229)
      at ko (node_modules/ts-auto-mock/merge/index.js:1:41155)
      at node_modules/ts-auto-mock/merge/index.js:1:41271
      at node_modules/ts-auto-mock/merge/index.js:1:38394
      at t (node_modules/ts-auto-mock/merge/index.js:1:41229)
      at ko (node_modules/ts-auto-mock/merge/index.js:1:41155)
      at node_modules/ts-auto-mock/merge/index.js:1:41271
      at node_modules/ts-auto-mock/merge/index.js:1:38394
      at t (node_modules/ts-auto-mock/merge/index.js:1:41229)
      at ko (node_modules/ts-auto-mock/merge/index.js:1:41155)
      at node_modules/ts-auto-mock/merge/index.js:1:41271
      at node_modules/ts-auto-mock/merge/index.js:1:38394
      at t (node_modules/ts-auto-mock/merge/index.js:1:41229)
      at ko (node_modules/ts-auto-mock/merge/index.js:1:41155)
      at node_modules/ts-auto-mock/merge/index.js:1:41271
      at node_modules/ts-auto-mock/merge/index.js:1:38394
      at t (node_modules/ts-auto-mock/merge/index.js:1:41229)
      at ko (node_modules/ts-auto-mock/merge/index.js:1:41155)
      at node_modules/ts-auto-mock/merge/index.js:1:41271
      at node_modules/ts-auto-mock/merge/index.js:1:38394
      at t (node_modules/ts-auto-mock/merge/index.js:1:41229)
      at ko (node_modules/ts-auto-mock/merge/index.js:1:41155)
      at node_modules/ts-auto-mock/merge/index.js:1:41271
      at node_modules/ts-auto-mock/merge/index.js:1:38394
      at t (node_modules/ts-auto-mock/merge/index.js:1:41229)
      at ko (node_modules/ts-auto-mock/merge/index.js:1:41155)
      at node_modules/ts-auto-mock/merge/index.js:1:41271
      at node_modules/ts-auto-mock/merge/index.js:1:38394
      at t (node_modules/ts-auto-mock/merge/index.js:1:41229)
      at ko (node_modules/ts-auto-mock/merge/index.js:1:41155)
      at node_modules/ts-auto-mock/merge/index.js:1:41271
      at node_modules/ts-auto-mock/merge/index.js:1:38394
      at t (node_modules/ts-auto-mock/merge/index.js:1:41229)
      at ko (node_modules/ts-auto-mock/merge/index.js:1:41155)
      at node_modules/ts-auto-mock/merge/index.js:1:41271
      at node_modules/ts-auto-mock/merge/index.js:1:38394
      at t (node_modules/ts-auto-mock/merge/index.js:1:41229)
      at ko (node_modules/ts-auto-mock/merge/index.js:1:41155)
      at node_modules/ts-auto-mock/merge/index.js:1:41271
      at node_modules/ts-auto-mock/merge/index.js:1:38394
      at t (node_modules/ts-auto-mock/merge/index.js:1:41229)
      at ko (node_modules/ts-auto-mock/merge/index.js:1:41155)
      at node_modules/ts-auto-mock/merge/index.js:1:41271
      at node_modules/ts-auto-mock/merge/index.js:1:38394
      at t (node_modules/ts-auto-mock/merge/index.js:1:41229)
      at ko (node_modules/ts-auto-mock/merge/index.js:1:41155)
      at node_modules/ts-auto-mock/merge/index.js:1:41271
      at node_modules/ts-auto-mock/merge/index.js:1:38394
      at t (node_modules/ts-auto-mock/merge/index.js:1:41229)
      at ko (node_modules/ts-auto-mock/merge/index.js:1:41155)
      at node_modules/ts-auto-mock/merge/index.js:1:41271
      at node_modules/ts-auto-mock/merge/index.js:1:38394
      at t (node_modules/ts-auto-mock/merge/index.js:1:41229)
      at ko (node_modules/ts-auto-mock/merge/index.js:1:41155)
      at node_modules/ts-auto-mock/merge/index.js:1:41271
      at node_modules/ts-auto-mock/merge/index.js:1:38394
      at t (node_modules/ts-auto-mock/merge/index.js:1:41229)
      at ko (node_modules/ts-auto-mock/merge/index.js:1:41155)
      at node_modules/ts-auto-mock/merge/index.js:1:41271
      at node_modules/ts-auto-mock/merge/index.js:1:38394
      at t (node_modules/ts-auto-mock/merge/index.js:1:41229)
      at ko (node_modules/ts-auto-mock/merge/index.js:1:41155)
      at node_modules/ts-auto-mock/merge/index.js:1:41271
      at node_modules/ts-auto-mock/merge/index.js:1:38394
      at t (node_modules/ts-auto-mock/merge/index.js:1:41229)
      at ko (node_modules/ts-auto-mock/merge/index.js:1:41155)
      at node_modules/ts-auto-mock/merge/index.js:1:41271
      at node_modules/ts-auto-mock/merge/index.js:1:38394
      at t (node_modules/ts-auto-mock/merge/index.js:1:41229)
      at ko (node_modules/ts-auto-mock/merge/index.js:1:41155)
      at node_modules/ts-auto-mock/merge/index.js:1:41271
      at node_modules/ts-auto-mock/merge/index.js:1:38394
      at t (node_modules/ts-auto-mock/merge/index.js:1:41229)
      at ko (node_modules/ts-auto-mock/merge/index.js:1:41155)
      at node_modules/ts-auto-mock/merge/index.js:1:41271
      at node_modules/ts-auto-mock/merge/index.js:1:38394
      at t (node_modules/ts-auto-mock/merge/index.js:1:41229)
      at ko (node_modules/ts-auto-mock/merge/index.js:1:41155)
      at node_modules/ts-auto-mock/merge/index.js:1:41271
      at node_modules/ts-auto-mock/merge/index.js:1:38394
      at t (node_modules/ts-auto-mock/merge/index.js:1:41229)
      at ko (node_modules/ts-auto-mock/merge/index.js:1:41155)
      at node_modules/ts-auto-mock/merge/index.js:1:41271
      at node_modules/ts-auto-mock/merge/index.js:1:38394
      at t (node_modules/ts-auto-mock/merge/index.js:1:41229)
      at ko (node_modules/ts-auto-mock/merge/index.js:1:41155)
      at node_modules/ts-auto-mock/merge/index.js:1:41271
      at node_modules/ts-auto-mock/merge/index.js:1:38394
      at t (node_modules/ts-auto-mock/merge/index.js:1:41229)
      at ko (node_modules/ts-auto-mock/merge/index.js:1:41155)
      at node_modules/ts-auto-mock/merge/index.js:1:41271
      at node_modules/ts-auto-mock/merge/index.js:1:38394
      at t (node_modules/ts-auto-mock/merge/index.js:1:41229)
      at ko (node_modules/ts-auto-mock/merge/index.js:1:41155)
      at node_modules/ts-auto-mock/merge/index.js:1:41271
      at node_modules/ts-auto-mock/merge/index.js:1:38394
      at t (node_modules/ts-auto-mock/merge/index.js:1:41229)
      at ko (node_modules/ts-auto-mock/merge/index.js:1:41155)
      at node_modules/ts-auto-mock/merge/index.js:1:41271
      at node_modules/ts-auto-mock/merge/index.js:1:38394
      at t (node_modules/ts-auto-mock/merge/index.js:1:41229)
      at ko (node_modules/ts-auto-mock/merge/index.js:1:41155)
      at node_modules/ts-auto-mock/merge/index.js:1:41271
      at node_modules/ts-auto-mock/merge/index.js:1:38394`

**Comment by uittorio** [comment]
@MikeAwie Hi. ! Sorry If I didn't reply earlier. Could you give me the declaration for the 2 interfaces? Model1 and Model2?

**Comment by magnusart** [comment]
I'm seeing a similar error, I cant say 100% sure it is the same error but it looks very similar.

My code looks like this
```typescript
    import admin from "firebase-admin"
    <code omitted...>
    const appMock: admin.app.App = createMock<admin.app.App>()

    const getFirestoreMock = On(appMock).get(method(mock => mock.firestore))
    const firestoreMock = createMock<admin.firestore.Firestore>()
    getFirestoreMock.mockImplementation(() => firestoreMock)
```

The output is
```bash
 RangeError: Maximum call stack size exceeded

      at resolveNameHelper (node_modules/typescript/lib/typescript.js:35490:35)
      at resolveName (node_modules/typescript/lib/typescript.js:35488:20)
      at resolveEntityName (node_modules/typescript/lib/typescript.js:36663:26)
      at getSymbolOfNameOrPropertyAccessExpression (node_modules/typescript/lib/typescript.js:65488:24)
      at getSymbolAtLocation (node_modules/typescript/lib/typescript.js:65535:28)
      at Object.getSymbolAtLocation (node_modules/typescript/lib/typescript.js:34483:31)
      at Object.r [as GetDeclarationFromNode] (node_modules/ts-auto-mock/transformer/index.js:1:5334)
      at node_modules/ts-auto-mock/transformer/index.js:1:37146
          at Array.reduce (<anonymous>)
      at Object.addFromDeclarationExtension (node_modules/ts-auto-mock/transformer/index.js:1:37053)
```

**Comment by uittorio** [comment]
I @magnusart , thanks for the report. In your scenario, the issue is related to this one. #113 
Not sure when it will be fixed, it's not an easy one :)

**Comment by uittorio** [comment]
Hi @magnusart this issue should be now fixed :) you can try to use the new version 2.0.0 of ts-auto-mock.

**Comment by uittorio** [comment]
Hi @MikeAwie, this should be fixed. I'll close for inactivity
